rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -159,18 +159,17 @@ object Reduce {
       ).scanLeft(0)(_ + _)
 
       def handle[A](
-          evalImpl: ((A, Blake2b512Random) => M[Unit]),
-          start: Int
+          evalImpl: ((A, Blake2b512Random) => M[Unit])
       )(input: (A, Int)): M[Unit] = {
         val (term, idx) = input
 
         val newRand =
           if (starts(6) == 1)
             rand
           else if (starts(6) > 256)
-            rand.splitShort((start + idx).toShort)
+            rand.splitShort((starts(idx) + idx).toShort)
           else
-            rand.splitByte((start + idx).toByte)
+            rand.splitByte((starts(idx) + idx).toByte)
```

Comment:
I'm not sure why this change is correct. Can you explain?

Reply:
It's not correct. I committed this without running the tests and I only realized it after I did the next step. Then I didn't undo because the fix was easy. See the commit "fix the tests"

Reply:
Yeah, as I mentioned - that commit could've easily be a `fixup! [Original commit message]`, squashed automatically upon rebase (provided you have `git config --global rebase.autosquash true`)